pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java

```diff
@@ -77,6 +78,8 @@
     protected AbstractMetadataStore() {
         this.executor = Executors
                 .newSingleThreadScheduledExecutor(new DefaultThreadFactory("metadata-store"));
+        this.listenerExecutor = Executors
+                .newSingleThreadScheduledExecutor(new DefaultThreadFactory("metadata-store-listener"));
```

Comment:
I'm not sure this is the right approach because it just moves the problem to a different thread. Instead we should make sure that no blocking calls are happening from the callbacks.

Reply:
In particular, this deadlock should already be fixed by #13885

Reply:
Thanks for that reference Matteo. Is there any value in adding an extra thread just in case a blocking method is inadvertently added back onto this calling thread? The risk seems high, considering we'd be blocking the zk notification handling thread.

Reply:
In most cases, I think the added thread is just moving the problem somewhere else but will not avoid the deadlock scenarios.

Reply:
I agree with @merlimat